<!DOCTYPE html
  PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
  "DTD/xhtml1-strict.dtd">
<html>
  <head>
    <title>twisted.trial._synctest._AssertRaisesContext : API documentation</title>
    <meta content="text/html;charset=utf-8" http-equiv="Content-Type" />
    <link href="apidocs.css" type="text/css" rel="stylesheet" />
    
    
  </head>
  <body>

    <span style="display: none" id="current-docs-container">
      <em>
        <a id="current-docs-link">
          Go to the latest version of this document.
        </a>
      </em>
    </span>

    <h1 class="class private">t.t._._AssertRaisesContext(<span title="object">object</span>) : class documentation</h1>
    <p>
      <span id="part">Part of <a href="twisted.html">twisted</a>.<a href="twisted.trial.html">trial</a>.<a href="twisted.trial._synctest.html">_synctest</a></span>
      <a href="http://twistedmatrix.com/trac/browser/tags/releases/twisted-15.0.0/twisted/trial/_synctest.py#L273">View Source</a>
      <a href="classIndex.html#twisted.trial._synctest._AssertRaisesContext">View In Hierarchy</a>
    </p>
    <div>
      
    </div>
    <div>A helper for implementing <code>assertRaises</code>.  This is a context 
manager and a helper method to support the non-context manager version of 
<code>assertRaises</code>.<table class="fieldTable"></table></div>

    
    
    <div id="splitTables">
      <table class="children sortable" id="id3092">
  
  
<tr class="instancevariable">
    
    
    <td>Instance Variable</td>
    <td><a href="twisted.trial._synctest._AssertRaisesContext.html#exception">exception</a></td>
    <td>The exception which was raised by the function being tested (if it raised 
one).</td>
  </tr><tr class="method">
    
    
    <td>Method</td>
    <td><a href="twisted.trial._synctest._AssertRaisesContext.html#__init__">__init__</a></td>
    <td></td>
  </tr><tr class="method">
    
    
    <td>Method</td>
    <td><a href="twisted.trial._synctest._AssertRaisesContext.html#__enter__">__enter__</a></td>
    <td><span class="undocumented">Undocumented</span></td>
  </tr><tr class="method">
    
    
    <td>Method</td>
    <td><a href="twisted.trial._synctest._AssertRaisesContext.html#__exit__">__exit__</a></td>
    <td><span>Check exit exception against expected exception.</span></td>
  </tr><tr class="instancevariable private">
    
    
    <td>Instance Variable</td>
    <td><a href="twisted.trial._synctest._AssertRaisesContext.html#_testCase">_testCase</a></td>
    <td>See <code>testCase</code> parameter of <code>__init__</code></td>
  </tr><tr class="instancevariable private">
    
    
    <td>Instance Variable</td>
    <td><a href="twisted.trial._synctest._AssertRaisesContext.html#_expected">_expected</a></td>
    <td>See <code>expected</code> parameter of <code>__init__</code></td>
  </tr><tr class="instancevariable private">
    
    
    <td>Instance Variable</td>
    <td><a href="twisted.trial._synctest._AssertRaisesContext.html#_returnValue">_returnValue</a></td>
    <td>The value returned by the callable being tested (only when not being used 
as a context manager).</td>
  </tr><tr class="instancevariable private">
    
    
    <td>Instance Variable</td>
    <td><a href="twisted.trial._synctest._AssertRaisesContext.html#_expectedName">_expectedName</a></td>
    <td>A short string describing the expected exception (usually the name of the 
exception class).</td>
  </tr><tr class="method private">
    
    
    <td>Method</td>
    <td><a href="twisted.trial._synctest._AssertRaisesContext.html#_handle">_handle</a></td>
    <td><span>Call the given object using this object as a context manager.</span></td>
  </tr>
  
</table>
      
      
    </div>
    
    
    

    <div class="function">
  <a name="twisted.trial._synctest._AssertRaisesContext._testCase">
    
  </a>
  <a name="_testCase">
    
  </a>
  <div class="functionHeader">
    _testCase =
  </div>
  <div class="functionBody">
    See <code>testCase</code> parameter of <code>__init__</code>
  </div>
</div><div class="function">
  <a name="twisted.trial._synctest._AssertRaisesContext._expected">
    
  </a>
  <a name="_expected">
    
  </a>
  <div class="functionHeader">
    _expected =
  </div>
  <div class="functionBody">
    See <code>expected</code> parameter of <code>__init__</code>
  </div>
</div><div class="function">
  <a name="twisted.trial._synctest._AssertRaisesContext._returnValue">
    
  </a>
  <a name="_returnValue">
    
  </a>
  <div class="functionHeader">
    _returnValue =
  </div>
  <div class="functionBody">
    The value returned by the callable being tested (only when not being used 
as a context manager).
  </div>
</div><div class="function">
  <a name="twisted.trial._synctest._AssertRaisesContext._expectedName">
    
  </a>
  <a name="_expectedName">
    
  </a>
  <div class="functionHeader">
    _expectedName =
  </div>
  <div class="functionBody">
    A short string describing the expected exception (usually the name of the 
exception class).
  </div>
</div><div class="function">
  <a name="twisted.trial._synctest._AssertRaisesContext.exception">
    
  </a>
  <a name="exception">
    
  </a>
  <div class="functionHeader">
    exception =
  </div>
  <div class="functionBody">
    The exception which was raised by the function being tested (if it raised 
one).
  </div>
</div><div class="function">
  <a name="twisted.trial._synctest._AssertRaisesContext.__init__">
    
  </a>
  <a name="__init__">
    
  </a>
  <div class="functionHeader">
    
    def
    __init__(self, testCase, expected):
    <a href="http://twistedmatrix.com/trac/browser/tags/releases/twisted-15.0.0/twisted/trial/_synctest.py#L293" class="functionSourceLink">
      
      (source)
    </a>
  </div>
  <div class="functionBody">
    
    <div><table class="fieldTable"><tr class="fieldStart"><td class="fieldName">Parameters</td><td class="fieldArg">testCase</td><td>The <a 
href="twisted.trial.unittest.TestCase.html"><code>TestCase</code></a> 
instance which is used to raise a test-failing exception when that is 
necessary.</td></tr><tr><td></td><td class="fieldArg">expected</td><td>The exception type expected to be raised.</td></tr></table></div>
  </div>
</div><div class="function">
  <a name="twisted.trial._synctest._AssertRaisesContext._handle">
    
  </a>
  <a name="_handle">
    
  </a>
  <div class="functionHeader">
    
    def
    _handle(self, obj):
    <a href="http://twistedmatrix.com/trac/browser/tags/releases/twisted-15.0.0/twisted/trial/_synctest.py#L309" class="functionSourceLink">
      
      (source)
    </a>
  </div>
  <div class="functionBody">
    
    <div>Call the given object using this object as a context manager.<table class="fieldTable"><tr class="fieldStart"><td class="fieldName">Parameters</td><td class="fieldArg">obj</td><td>The object to call and which is expected to raise some exception. (type: <a 
href="http://docs.python.org/library/functions.html#object"><code>object</code></a>)</td></tr><tr class="fieldStart"><td class="fieldName">Returns</td><td colspan="2">Whatever exception is raised by <code>obj()</code>. (type: <a 
href="http://docs.python.org/library/exceptions.html#exceptions.BaseException"><code>BaseException</code></a>)</td></tr></table></div>
  </div>
</div><div class="function">
  <a name="twisted.trial._synctest._AssertRaisesContext.__enter__">
    
  </a>
  <a name="__enter__">
    
  </a>
  <div class="functionHeader">
    
    def
    __enter__(self):
    <a href="http://twistedmatrix.com/trac/browser/tags/releases/twisted-15.0.0/twisted/trial/_synctest.py#L325" class="functionSourceLink">
      
      (source)
    </a>
  </div>
  <div class="functionBody">
    
    <div class="undocumented">Undocumented</div>
  </div>
</div><div class="function">
  <a name="twisted.trial._synctest._AssertRaisesContext.__exit__">
    
  </a>
  <a name="__exit__">
    
  </a>
  <div class="functionHeader">
    
    def
    __exit__(self, exceptionType, exceptionValue, traceback):
    <a href="http://twistedmatrix.com/trac/browser/tags/releases/twisted-15.0.0/twisted/trial/_synctest.py#L329" class="functionSourceLink">
      
      (source)
    </a>
  </div>
  <div class="functionBody">
    
    <div>Check exit exception against expected exception.<table class="fieldTable"></table></div>
  </div>
</div>
    <address>
      <a href="index.html">API Documentation</a> for <a href="http://twistedmatrix.com/">Twisted</a>, generated by <a href="http://codespeak.net/~mwh/pydoctor/">pydoctor</a> at 2015-01-30 14:14:53.
    </address>

    <!-- Google analytics, obviously. -->
    <script src="http://www.google-analytics.com/urchin.js" type="text/javascript"></script>
    <script type="text/javascript">
      _uacct = "UA-99018-6";
      urchinTracker();
    </script>

    <!-- If the documentation isn't current, insert a current link. -->
    <script type="text/javascript">
      if (window.location.pathname.indexOf('/current/') == -1) {
          <!-- Give the user a link to this page, but in the current version of the docs. -->
          var link = document.getElementById('current-docs-link');
          link.href = window.location.pathname.replace(/\/\d+\.\d+\.\d+\/api\//, '/current/api/');
          <!-- And make it visible -->
          var container = document.getElementById('current-docs-container');
          container.style.display = "";
          delete link;
          delete container;
      }
    </script>

  </body>
</html>